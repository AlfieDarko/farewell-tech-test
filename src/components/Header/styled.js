import styled from "styled-components"
import { COLOR, SPACING, CONTENT_WIDTH } from "../../constants"

export const Header = styled.header`
  background-color: ${COLOR.ACCENT.PRIMARY};
  padding: ${SPACING.XS} ${SPACING.M};
`

export const HeaderWrapper = styled.div`
  margin: 0 auto;
  max-width: ${CONTENT_WIDTH};
`

export const HeaderContent = styled.div`
  align-items: center;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
`
