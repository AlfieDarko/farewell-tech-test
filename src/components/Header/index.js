import React from "react"
import PropTypes from "prop-types"
import Logo from "../../logo.svg"
import * as Styled from "./styled"

const Header = ({ dataTestId }) => (
  <Styled.Header data-testid={dataTestId}>
    <Styled.HeaderWrapper>
      <Styled.HeaderContent>
        <img src={Logo} alt="Farewill" />
        <p>Simpsons Quotes</p>
      </Styled.HeaderContent>
    </Styled.HeaderWrapper>
  </Styled.Header>
)

Header.propTypes = {
  dataTestId: PropTypes.string.isRequired,
}

export default Header
