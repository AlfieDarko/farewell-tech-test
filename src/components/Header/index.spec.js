import React from "react"
import { render, screen } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect"
import Header from "./index"

describe("Button Component", () => {
  it("Should find Button Component by its data attribute", () => {
    render(<Header dataTestId={"header"} />)
    expect(screen.getByTestId("header")).toBeInTheDocument()
  })

  it("Should match the snapshot", () => {
    const { asFragment } = render(
      <Header data-testid={"header"}>"Hello"</Header>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
