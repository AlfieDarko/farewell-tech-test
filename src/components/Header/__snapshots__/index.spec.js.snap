// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Should match the snapshot 1`] = `
<DocumentFragment>
  <header
    class="sc-bdnylx jlFsoE"
  >
    <div
      class="sc-gtssRu jmGQPz"
    >
      <div
        class="sc-dlnjPT epcXMg"
      >
        <img
          alt="Farewill"
          src="logo.svg"
        />
        <p>
          Simpsons Quotes
        </p>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
